### PR TITLE
Easier configuration and usage of composer library 

### PR DESCRIPTION
**Pull request by tsari**
(add) | profiler options which are already implemented in perftools/xhgui
(add) | env vars for some configuration entries to make a use with docker more pleasant
(add) | introduce XHGUI_CONFIG_DIR constant as way to set the location of your xhgui config.php
(update) | readme

With these changes your are able to set your own config dir when requiring the header.php in the application.
You can run your application in docker container and start the profiling via env var or set the an env var to change the profiling rate.

### CODE REVIEW

File: config/config.default.php [hunk 1]
```diff
@@ -2,6 +2,11 @@
 /**
  * Default configuration for Xhgui
  */
+
+$mongoUri = getenv('XHGUI_MONGO_URI') ?: '127.0.0.1:27017';
+$mongoUri = str_replace('mongodb://', '', $mongoUri);
+$mongoDb  = getenv('XHGUI_MONGO_DB') ?: 'xhprof';
```
Comment (markstory) [inline config/config.default.php]: Not a fan of space aligning as it complicates future maintenance.

File: config/config.default.php [hunk 2]
```diff
@@ -27,14 +32,17 @@
     'detail.count' => 6,
     'page.limit' => 25,
 
-    // Profile 1 in 100 requests.
+    // Profile x in 100 requests. (E.g. set XHGUI_PROFLING_RATIO=50 to profile 50% of requests)
     // You can return true to profile every request.
     'profiler.enable' => function() {
-        return rand(1, 100) === 42;
+        $ratio = getenv('XHGUI_PROFILING_RATIO') ?: 100;
+        return (getenv('XHGUI_PROFILING') !== false) && (mt_rand(1, 100) <= $ratio);
```
Comment (markstory) [inline config/config.default.php:39]: When would this be `false`? Environment variables are string typed so `XHGUI_PROFILING=false` will result in `'false'` as a string.

**Review by markstory — COMMENTED** [review]
Looking good. I like the use of environment variables :clap:

File: config/config.default.php (hunk 2 above)
Comment (tsari) [reply config/config.default.php:39]: `getenv()` returns `false` if the environment var is not set. So profiling is disabled by default. If you set the environment var with any content, the profiling will be activated.

Comment (markstory) [reply config/config.default.php:39]: Ok.

Comment (tsari) [reply config/config.default.php:39]: Do you merge this into the master? If so, when? I need this for the company I work for.

**Comment by ossinkine** [comment]
@tsari Hi there, I've noticed that before this PR the bihavior of config loading was different. The default config is loaded first and then the customized config is loaded. This allowed us to describe only the changed value, and not all. The XHGui behavior is still [the same](https://github.com/perftools/xhgui/blob/master/src/bootstrap.php#L13-L16). Why was this changed here?

**Pull request merged**